src/renderer/components/message/MessageList.js

```diff
@@ -75,17 +75,6 @@ export default function MessageList ({ chat, refComposer, locationStreamingEnabl
     scrollToBottom()
   }, [chat.id])
 
-  const onClickAttachment = (message) => {
-    if (message.viewType.viewType === 23) return
-    if (isDisplayableByRenderMedia(message.msg.attachment)) {
-      openDialog('RenderMedia', { message })
-    } else {
-      if (!shell.openItem(message.msg.attachment.url)) {
-        logger.info("file couldn't be opened, try saving it in a different place and try to open it from there")
-      }
-    }
-  }
-
   const onClickSetupMessage = setupMessage => openDialog('EnterAutocryptSetupMessage', { setupMessage })
   const onShowDetail = message => openDialog('MessageDetail', { message, chat })
```

Comment:
why do we have to reuse the chat object here? I would appreciate it when that dialog could fetch it from the core again, that would allow us to move this function around and inside of the message.

Reply:
same on delete message, the dependency to the chatStore is annoying

Reply:
Also I would appreciate if we could call openDialog without a ScreenContext.

Reply:
> 

why do we have to reuse the chat object here? I would appreciate it when that dialog could fetch it from the core again, that would allow us to move this function around and inside of the message.


The message object should be enough for both? I think we shouldn't refetch anything that we already have, also because this could introduce inconsistencies if something changed in between. ChatStore is annoying anyways... openDialog without a screenContext sounds not so easy for me, how do you want to achieve this? Global variable?